sdk/messaging/azservicebus/messageSettler.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azservicebus
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal"
+	"github.com/Azure/go-amqp"
+)
+
+var errReceiveAndDeleteReceiver = errors.New("messages that are received in receiveAndDelete mode are not settlable")
+
+type messageSettler struct {
```

Comment:
Should this go under `internal`, how are we making that distinction?

Reply:
I'm not saying it should, just wondering how you're thinking about when things should/shouldn't.

Reply:
Maybe there's a more general pattern to distill here and share across the team.

Reply:
This is a goofy one.

Yes, I wanted to put under `internal` _but_ it references `Message`, which is in the public surface. So that caused a circular reference.

Is there a way to work around that aside from creating a common package between the two? I didn't want to do all of that in this PR, but I did have the same desire.

Reply:
Ah I see your point.  I don't think there's a good way to achieve this separation outside of what you recommended, however I'm not sure if the work is really worth it given the narrow scope of this type (well, I assume the scope is narrow but maybe I'm mistaken).  As long as we follow our rule about only using "exported" fields/methods it's probably good enough (I wonder if there's a linter rule for this....).

Reply:
For now I'll leave it. It might be nice to rework it a bit because the top level of this package is stuffed with files and it'd be nice if it was only private stuff.